src/core/comparable.ml

```diff
@@ -5,3 +5,9 @@ module type S = sig
 
   val compare:t Comparator.t
 end
+
+module type C = sig
```

Comment:
Hi yeah, I have read the paper, I agree this interface is a bit unusual. I took the idea from Core, but it seems Haskell does not have the notation, nor any other language I can recall. I will remove it then.